Sources/Classes/RSClient.m

```diff
@@ -225,23 +225,32 @@ - (void)alias:(NSString *)newId {
         [self reportDiscardedEvent];
         return;
     }
-    [self alias:newId options:nil];
+    [self alias:newId options:nil previousId:nil];
 }
 
 - (void) alias:(NSString *)newId options:(RSOption *) options {
+    [self alias:newId options:options previousId:nil];
+}
+
+- (void) alias:(NSString *)newId options:(RSOption *) options previousId:(NSString *)previousId {
     if ([RSClient getOptStatus]) {
         [self reportDiscardedEvent];
         return;
     }
     RSContext *rc = [RSElementCache getContext];
     NSMutableDictionary<NSString*,NSObject*>* traits = [rc.traits mutableCopy];
     
-    NSObject *prevId = [traits objectForKey:@"userId"];
-    if(prevId == nil) {
-        prevId =[traits objectForKey:@"id"];
-    }
-    if (prevId == nil) {
-        prevId = self.anonymousId;
+    NSObject *prevId = nil;
+    if (previousId != nil) {
+        prevId = previousId;
+    } else {
+        prevId = [traits objectForKey:@"userId"];
+        if(prevId == nil) {
+            prevId =[traits objectForKey:@"id"];
+        }
+        if (prevId == nil) {
+            prevId = self.anonymousId;
+        }
```

Comment:
To avoid multiple if cases, use Ternary Conditional Operator(?:) like below.

`NSObject *prevId = previousId ?: [traits objectForKey:@"userId"] ?: [traits objectForKey:@"id"] ?: self.anonymousId;`